### PR TITLE
[23.05] nixos/users-groups: add new xdg-compliant paths to `environment.profiles`

### PR DESCRIPTION
**Pull request by Ma27**
## Description of changes
Effectively #241518 & #260764. This is also relevant for 23.05 because Nix 2.17 is available there as well!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/config/users-groups.nix [hunk 1]
```diff
@@ -686,6 +686,8 @@ in {
 
     environment.profiles = [
       "$HOME/.nix-profile"
+      "\${XDG_STATE_HOME}/nix/profile"
```
Comment (bjornfor) [inline nixos/modules/config/users-groups.nix:689]: Why is the XDG_STATE_HOME line using a different environment expansion syntax?

Comment (Ma27) [reply nixos/modules/config/users-groups.nix:689]: @bjornfor wdym?

Comment (Artturin) [reply nixos/modules/config/users-groups.nix:689]: $HOME is actually special-cased and converted to `${HOME}` in https://github.com/NixOS/nixpkgs/blob/f99e5f03cc0aa231ab5950a15ed02afec45ed51a/nixos/modules/config/system-environment.nix#L75

Comment (bjornfor) [reply nixos/modules/config/users-groups.nix:689]: Ok, thanks.

**Comment by Ma27** [comment]
@bjornfor @Artturin anything else to be fixed? Or do you think we could merge this? %)

**Pull request merged**